src/MfaVerifyController.js

```diff
@@ -89,12 +89,12 @@ function (Okta, Checkbox, BaseLoginController, CookieUtil, TOTPForm, YubikeyForm
         }
 
         // Remember Device checkbox resides outside of the Push and TOTP forms.
-        if (this.settings.get('features.rememberDevice')) {
+        if (this.settings.get('features.rememberDevice') && this.options.appState.get('allowRememberDevice')) {
```

Comment:
Also, since we are using this - `this.settings.get('features.rememberDevice') && this.options.appState.get('allowRememberDevice')` everywhere, it might make sense to make this a model attribute or a helper function and use it everywhere.